### PR TITLE
[update] matrix.rst

### PR DESCRIPTION
**Pull request by isdanni**
### file changed
- `matrix.po`

### related issue

#101

### CODE REVIEW

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 1]
```diff
@@ -279,155 +306,84 @@ msgstr "（矩阵）该矩阵本身"
...
 
 #: ../../matrix.rst:152
 msgid ""
 "a = ti.Matrix([[2, 3], [4, 5]])\n"
 "b = a.transpose()\n"
 "# Now b = ti.Matrix([[2, 4], [3, 5]])"
 msgstr ""
+"a = ti.Matrix([[2, 3], [4, 5]])\n"
+"b = a.transpose()\n"
+"# Now b = ti.Matrix([[2, 4], [3, 5]])"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
"# 现在 b = ti.Matrix([[2, 4], [3, 5]])"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 2]
```diff
@@ -279,155 +306,84 @@ msgstr "（矩阵）该矩阵本身"
...
-"result."
-msgstr ""
+"``a.transpose()`` will not effect the data in ``a``, it just return the result."
+msgstr "``a.transpose()`` 不会影响 ``a`` 中的数据，它只是返回结果。"
 
 #: ../../matrix.rst:164
-#, fuzzy
 msgid "(scalar) the trace of matrix ``a``."
-msgstr "（标量）矩阵的行数"
+msgstr "（标量） ``a`` 矩阵的痕迹。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "（标量） ``a`` 矩阵的迹。"
```
I thought it's a standard translation in many linalg textbooks... let's follow them.

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 3]
```diff
@@ -279,155 +306,84 @@ msgstr "（矩阵）该矩阵本身"
...
 #: ../../matrix.rst:166
 msgid "The return value can be computed as ``a[0, 0] + a[1, 1] + ...``."
-msgstr ""
+msgstr "返回值可以计算为 ``a[0, 0] + a[1, 1] + ...`` 。"
 
 #: ../../matrix.rst:172
-#, fuzzy
 msgid "(scalar) the determinant of matrix ``a``."
-msgstr "（标量）矩阵的行数"
+msgstr "（标量） ``a`` 矩阵的决定因素。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "（标量） ``a`` 矩阵的行列式。"
```
Please don't depend too much on machine translators :)

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 4]
```diff
@@ -279,155 +306,84 @@ msgstr "（矩阵）该矩阵本身"
...
 
 #: ../../matrix.rst:176 ../../matrix.rst:188
 msgid "The matrix size of matrix must be 1x1, 2x2, 3x3 or 4x4 for now."
-msgstr ""
+msgstr "矩阵的矩阵大小现在必须为 1x1、2x2、3x3 或 4x4。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "目前用于此函数的矩阵大小必须为 1x1、2x2、3x3 或 4x4。"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 5]
```diff
@@ -279,155 +306,84 @@ msgstr "（矩阵）该矩阵本身"
...
 
 #: ../../matrix.rst:178 ../../matrix.rst:190
 msgid "This function only works in Taichi-scope for now."
-msgstr ""
+msgstr "此功能目前仅在Taichi范围内有效。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "此函数目前仅在 Taichi 作用域内有效。"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 6]
```diff
@@ -279,155 +306,84 @@ msgstr "（矩阵）该矩阵本身"
...
 
 #: ../../matrix.rst:184
-#, fuzzy
 msgid "(Matrix) the inverse of matrix ``a``."
-msgstr "（矩阵）该矩阵本身"
+msgstr "（矩阵） ``a`` 的逆反矩阵。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "（矩阵） ``a`` 的逆矩阵。"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 7]
```diff
@@ -40,33 +42,34 @@ msgstr "注意区分逐元素的乘法 ``*`` 和矩阵乘法 ``@`` 。"
...
 
 #: ../../matrix.rst:13
 msgid "``any(A)`` (Taichi-scope only)"
-msgstr ""
+msgstr "``any(A)``（仅Taichi范围）"
 
 #: ../../matrix.rst:14
 msgid "``all(A)`` (Taichi-scope only)"
-msgstr ""
+msgstr "``all(A)``（仅Taichi范围）"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "``all(A)``（仅 Taichi 作用域）"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 8]
```diff
@@ -40,33 +42,34 @@ msgstr "注意区分逐元素的乘法 ``*`` 和矩阵乘法 ``@`` 。"
...
-msgstr "``U, sigma, V = ti.svd(A, ti.f32)``  （其中 ``sigma`` 是一个 ``3x3`` 矩阵）"
+"``U, sigma, V = ti.svd(A, ti.f32)`` (Note that ``sigma`` is a ``3x3`` diagonal "
+"matrix)"
+msgstr ""
+"``U, sigma, V = ti.svd(A, ti.f32)`` （其中 ``sigma`` 是一个 ``3x3`` 矩阵）"
 
 #: ../../matrix.rst:13
 msgid "``any(A)`` (Taichi-scope only)"
-msgstr ""
+msgstr "``any(A)``（仅Taichi范围）"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "``any(A)``（仅 Taichi 作用域）"
```

**Review by archibate — CHANGES_REQUESTED** [review]
Thank for the quick response! I added few nits :rocket:

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 9]
```diff
@@ -78,15 +81,17 @@ msgstr "Taichi中的矩阵有两种形式："
 
 #: ../../matrix.rst:20
 msgid ""
-"as a temporary local variable. An ``n by m`` matrix consists of ``n * m``"
-" scalar values."
+"as a temporary local variable. An ``n by m`` matrix consists of ``n * m`` "
+"scalar values."
 msgstr "作为临时本地变量。一个由 ``n*m`` 个标量构成的 ``n×m`` 阶矩阵。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "作为临时局部变量。一个由 ``n*m`` 个标量构成的 ``n×m`` 阶矩阵。"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 10]
```diff
@@ -113,13 +118,12 @@ msgid "(DataType) data type of the components"
 msgstr "（数据类型）元素的数据类型"
 
 #: ../../matrix.rst:34
-#, fuzzy
 msgid "(optional, scalar or tuple) shape the tensor of vectors, see :ref:`tensor`"
-msgstr "（标量或元组）向量张量的形状，见 :ref:`tensor`"
+msgstr "（可选、标量或元组）向量张量的形状，见 :ref:`tensor`"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "（可选，标量或元组）向量张量的形状，见 :ref:`tensor`"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 11]
```diff
@@ -113,13 +118,12 @@ msgid "(DataType) data type of the components"
...
 
 #: ../../matrix.rst:35
 msgid "(optional, scalar or tuple) see :ref:`offset`"
-msgstr ""
+msgstr "（可选、标量或元组）请参见 ： ref:`offset`"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "（可选，标量或元组）请参见 ： ref:`offset`"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 12]
```diff
@@ -160,17 +166,20 @@ msgid "(scalar) the second component of the second row"
...
 #: ../../matrix.rst:58
-#, fuzzy
 msgid ""
-"For example, this creates a 2x3 matrix with components (2, 3, 4) in the "
-"first row and (5, 6, 7) in the second row: ::"
-msgstr "比如，创建一个2x2的矩阵，第一行元素为 (2, 3)，第二行元素为 (4, 5)： ::"
+"For example, this creates a 2x3 matrix with components (2, 3, 4) in the first "
+"row and (5, 6, 7) in the second row: ::"
+msgstr ""
+"例如，这将创建一个 2x3 矩阵，在第一行中具有组件 (2, 3, 4) ，在第二行 (5, 6, 7) "
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
"例如，下述将创建一个 2x3 的矩阵，第一行中的分量为 (2, 3, 4) ，第二行的为 (5, 6, 7) "
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 13]
```diff
@@ -186,8 +195,8 @@ msgstr "（向量）组成第三行（列）元素的向量"
 
 #: ../../matrix.rst:72
 msgid ""
-"For example, this creates a 3x3 matrix by concactinating vectors into "
-"rows (or columns): ::"
+"For example, this creates a 3x3 matrix by concactinating vectors into rows (or "
+"columns): ::"
 msgstr "比如，以下代码通过将向量连接作为一个3x3矩阵的每一行（列）： ::"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "例如，以下代码创建了一个 3x3 矩阵，矩阵的行（列）是由向量拼接构成的： ::"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 14]
```diff
@@ -206,6 +215,19 @@ msgid ""
 "# lists can be used instead of vectors\n"
 "a = ti.Matrix.rows([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, 9.0]])"
 msgstr ""
+"# Taichi-scope\n"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
"# Taichi 作用域\n"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 15]
```diff
@@ -206,6 +215,19 @@ msgid ""
...
+"v0 = ti.Vector([1.0, 2.0, 3.0])\n"
+"v1 = ti.Vector([4.0, 5.0, 6.0])\n"
+"v2 = ti.Vector([7.0, 8.0, 9.0])\n"
+"\n"
+"# to specify data in rows\n"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
"# 指定行中的数据\n"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 16]
```diff
@@ -206,6 +215,19 @@ msgid ""
...
+"a = ti.Matrix.rows([v0, v1, v2])\n"
+"\n"
+"# to specify data in columns instead\n"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
"# 指定列中的数据\n"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 17]
```diff
@@ -206,6 +215,19 @@ msgid ""
...
+"a = ti.Matrix.cols([v0, v1, v2])\n"
+"\n"
+"# lists can be used instead of vectors\n"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
"# 可以用向量代替参数中的列表\n"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 18]
```diff
@@ -247,29 +269,34 @@ msgid ""
 "mat = a[6, 3]\n"
 "x = mat[0, 0]"
 msgstr ""
+"x = a[6, 3][0, 0]\n"
+"\n"
+"# or\n"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
"# 或者\n"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 19]
```diff
@@ -247,29 +269,34 @@ msgid ""
...
+"tensors of matrices."
 msgstr "访问矩阵形式张量中的标量元素时， **必须** 使用两对方括号。"
 
 #: ../../matrix.rst:116
 msgid ""
-"The indices in the first pair of brackets locate the matrix inside the "
-"tensor of matrices;"
+"The indices in the first pair of brackets locate the matrix inside the tensor "
+"of matrices;"
 msgstr "第一对方括号中的下标找到张量中的矩阵;"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "第一对方括号中的索引定位张量中的矩阵;"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 20]
```diff
@@ -247,29 +269,34 @@ msgid ""
...
 
 #: ../../matrix.rst:117
 msgid ""
-"The indices in the second pair of brackets locate the scalar element "
-"inside the matrix."
+"The indices in the second pair of brackets locate the scalar element inside the "
+"matrix."
 msgstr "第二对方括号中的下标找到矩阵中的标量元素。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "第二对方括号中的索引定位矩阵中的标量元素。"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 21]
```diff
@@ -247,29 +269,34 @@ msgid ""
...
 
 #: ../../matrix.rst:119
 msgid ""
-"For 0-D tensors of matrices, indices in the first pair of brackets should"
-" be ``[None]``."
+"For 0-D tensors of matrices, indices in the first pair of brackets should be "
+"``[None]``."
 msgstr "对于0维的矩阵形式的张量，第一对方括号中的下标应该为 ``[None]`` 。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "对于0维的矩阵形式的张量，第一对方括号中的索引应该为 ``[None]`` 。"
```

**Review by ArkhamWJZ — COMMENTED** [review]
LGTM! + A few nits 🌈

**Review by ArkhamWJZ — APPROVED** [review]

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 22]
```diff
@@ -113,13 +118,12 @@ msgid "(DataType) data type of the components"
...
 #: ../../matrix.rst:34
-#, fuzzy
 msgid "(optional, scalar or tuple) shape the tensor of vectors, see :ref:`tensor`"
-msgstr "（标量或元组）向量张量的形状，见 :ref:`tensor`"
+msgstr "（可选，标量或元组）向量张量的形状，见 :ref:`tensor`"
 
 #: ../../matrix.rst:35
 msgid "(optional, scalar or tuple) see :ref:`offset`"
-msgstr ""
+msgstr "（可选，标量或元组）请参见 ： ref:`offset`"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
msgstr "（可选，标量或元组）请参见 :ref:`offset`"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 23]
```diff
@@ -130,14 +134,16 @@ msgid ""
...
 msgid ""
-"In Python-scope, ``ti.var`` declares :ref:`scalar_tensor`, while "
-"``ti.Matrix`` declares tensors of matrices."
+"In Python-scope, ``ti.var`` declares :ref:`scalar_tensor`, while ``ti.Matrix`` "
+"declares tensors of matrices."
 msgstr ""
-"在 Python-scope 中， ``ti.var`` 声明了 :ref:`scalar_tensor` , ``ti.Matrix`` "
-"声明了矩阵形式的张量。"
+"在 Python-scope 中， ``ti.var`` 声明了 :ref:`scalar_tensor` , ``ti.Matrix`` 声明"
+"了矩阵形式的张量。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
"了由矩阵组成的张量。"
```

File: locales/zh_CN/LC_MESSAGES/matrix.po [hunk 24]
```diff
@@ -206,6 +215,19 @@ msgid ""
...
+"v1 = ti.Vector([4.0, 5.0, 6.0])\n"
+"v2 = ti.Vector([7.0, 8.0, 9.0])\n"
+"\n"
+"# 指定行中的数据\n"
+"a = ti.Matrix.rows([v0, v1, v2])\n"
+"\n"
+"# 指定列中的数据\n"
+"a = ti.Matrix.cols([v0, v1, v2])\n"
+"\n"
+"# 可以用向量代替参数中的列表\n"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/matrix.po]:
```suggestion
"# 可以用列表代替参数中的向量\n"
```

**Review by archibate — COMMENTED** [review]
Feel free to merge on your own after these minor issues are fixed :+1:

**Pull request merged**